community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/NeoStoreFileListing.java

```diff
@@ -64,17 +66,38 @@ public NeoStoreFileListing( File storeDir, LabelScanStore labelScanStore, Indexi
 
     public ResourceIterator<StoreFileMetadata> listStoreFiles( boolean includeLogs ) throws IOException
     {
-        Collection<StoreFileMetadata> files = new ArrayList<>();
+        List<StoreFileMetadata> files = new ArrayList<>();
         gatherNonRecordStores( files, includeLogs );
         gatherNeoStoreFiles( files );
         Resource labelScanStoreSnapshot = gatherLabelScanStoreFiles( files );
         Resource schemaIndexSnapshots = gatherSchemaIndexFiles( files );
         Resource legacyIndexSnapshots = gatherLegacyIndexFiles( files );
 
+        placeMetaDataStoreLast( files );
+
         return resourceIterator( files.iterator(),
                 new MultiResource( asList( labelScanStoreSnapshot, schemaIndexSnapshots, legacyIndexSnapshots ) ) );
     }
 
+    private void placeMetaDataStoreLast( List<StoreFileMetadata> files )
```

Comment:
It's used by `NeoStores.isStorePresent` to indicate that the store is present. Agreeably not very reliable.